### PR TITLE
Ignore reset action in stack navigator if reset key does not match current navigator

### PR DESCRIPTION
**Pull request by rpopovici**
Fall through if reset action key does not match navigation state.

Use case scenario:
In our app we have triple stack navigation structure like this:
Modal **StackNavigator** -> Home **TabNavigator** -> Child **StackNavigator**

With the current implementation, if you try to reset an **inactive** route in Home Tab Navigator by specifying a reset key to the reset action, it's not going to work because it will attempt to reset the active route of the tab navigator only.

Basically, the StackRouter is not able to ignore a reset action which is assigned to another stack router.

You can check this issue as well for more info https://github.com/react-community/react-navigation/issues/2291

### CODE REVIEW

**Comment by vonovak** [comment]
@rpopovici I'm afraid you need to improve the PR description, maybe add a demo or something to make people understand what this is about, because after reading the description I honestly have no idea

**Comment by rpopovici** [comment]
@vonovak hope this is better

**Comment by ssorallen** [comment]
Looks like this was superseded by https://github.com/react-navigation/react-navigation/pull/3400. That PR is almost exactly the same change but with negated logic.

**Comment by brentvatne** [comment]
indeed, thanks for pointing that out @ssorallen!

cc @ericvicenti
